core/templates/components/question-directives/question-player/skill-mastery-modal.component.ts

```diff
@@ -0,0 +1,61 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for skill mastery modal.
+ */
+
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+import { ConfirmOrCancelModal } from 'components/common-layout-directives/common-elements/confirm-or-cancel-modal.component';
+
+@Component({
+  selector: 'oppia-skill-mastery-modal',
+  templateUrl: './skill-mastery-modal.component.html'
+})
+export class SkillMasteryModalComponent
+  extends ConfirmOrCancelModal implements OnInit {
+  @Input() skillId: string = '';
```

Comment:
Why is this set to empty initially?

Reply:
![Screenshot from 2022-08-08 23-05-06](https://user-images.githubusercontent.com/77006897/183478946-919d0a6a-dbdb-44a6-b785-8f6d9cc2c784.png)

Reply:
we are passing skillId to another component so it has some value otherwise it though type error.